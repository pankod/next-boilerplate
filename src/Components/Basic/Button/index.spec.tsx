// #region Global Imports
import React from "react";
import { shallow, mount } from "enzyme";
// #endregion Global Imports

// #region Local Imports
import { Button } from "./index";
// #endregion Local Imports

describe("Basic Components", () => {
    describe("Button", () => {
        it("should have the passed class", () => {
            const wrapper = shallow(<Button className="active" />);

            expect(wrapper.hasClass("active")).toBe(true);
        });

        it("should render with children prop", () => {
            const wrapper = shallow(<Button>hey</Button>);

            expect(wrapper.props().children).toBe("hey");
        });

        it("should increment number on click", () => {
            let number = 1;
            const wrapper = shallow(
                <Button
                    onClick={() => {
                        number += 1;
                    }}
                />
            );

            wrapper.simulate("click");
            expect(number).toBe(2);
        });

        it("should match snapshot", () => {
            const wrapper = mount(<Button>Test</Button>);
            expect(wrapper).toMatchSnapshot();
        });
    });
});
