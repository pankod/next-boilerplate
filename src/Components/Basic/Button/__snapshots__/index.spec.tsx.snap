// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Components Button should match snapshot 1`] = `
.c0 {
  cursor: pointer;
}

<Component>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "
    cursor: pointer;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="c0"
      >
        Test
      </div>
    </StyledComponent>
  </styled.div>
</Component>
`;
